test_driver/app_test.dart

```diff
@@ -72,97 +80,169 @@ void main() {
 
-        // Close drawer menu
-        await driver.scroll(
-            drawerMenu, -300.0, 0.0, const Duration(milliseconds: 500));
+        // Open outdoor interest page
+        await tap(driver, outdoorInterest);
+
+        // SGW tab
+        await tap(driver, sgwTab);
+        await driver.waitFor(card);
+        await delay(1000);
+        await snapshot(driver, '\\OutdoorInterest\\SGW');
+
+        // Loyola tab
+        await tap(driver, loyolaTab);
+        await driver.waitFor(card);
+        await delay(1000);
+        await snapshot(driver, '\\OutdoorInterest\\Loyola');
       },
       timeout: Timeout(
         Duration(minutes: 1),
       ),
     );
 
     test(
-      'see current location and campus building highlights',
+      '[see current location and campus building highlights]',
       () async {
-        final swapCampusIcon = find.byValueKey('SwitchCampus');
-        final sun = find.byValueKey('ToggleBuildingHighlight');
-        await snapshot(driver, '\\MapLocation\\1');
+        // Back
+        await driver.tap(find.byTooltip('Back'));
 
-        // Toggle campus building hightlights
+        await snapshot(driver, '\\MapLocation\\1');
         await tap(driver, swapCampusIcon);
         await snapshot(driver, '\\MapLocation\\2');
-        await tap(driver, sun);
-        await snapshot(driver, '\\MapLocation\\3');
-        await tap(driver, sun);
         await tap(driver, swapCampusIcon);
-        await snapshot(driver, '\\MapLocation\\4');
+        await snapshot(driver, '\\MapLocation\\3');
       },
       timeout: Timeout(
         Duration(minutes: 1),
       ),
     );
 
     test(
-      'get directions',
+      '[get outdoor directions]',
       () async {
-        // Open search location page
+        // Open directions panel
         await tap(driver, search);
-        await snapshot(driver, '\\Directions\\1');
 
-        // Tap second suggested location
-        await tap(driver, location1);
-        await delay(3000);
+        // Pick start and end locations
+        await tap(driver, startLocation);
+        await delay(1000);
+        await driver.enterText("CENTRAL");
+        await delay(1000);
+        await tap(driver, location0);
 
-        // Slide open directions drawer
-        await driver.scroll(
-            driving, 0.0, -500.0, const Duration(milliseconds: 800));
+        await tap(driver, endLocation);
+        await delay(1000);
+        await driver.enterText("H913");
+        await delay(1000);
+        await tap(driver, location0);
+
+        // Wait to load
+        await delay(2000);
+        await snapshot(driver, '\\Directions\\DirectionsPanel');
 
         // Go through modes and expect at least first direction tile
+        await tap(driver, driving);
         await snapshot(driver, '\\Directions\\driving');
-        await driver.waitFor(direction1, timeout: Duration(seconds: 5));
+        await driver.waitFor(direction0, timeout: Duration(seconds: 5));
```

Comment:
Same thing here

Reply:
.waitFor will timeout eventually if it does find it (first direction of itinerary). If it doesn't appear, then there's either a network issue, a slow response, or an error. I'm making the assumption that if it doesn't show up in 5 sec then it won't ever show up.